llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1105,15 +1109,26 @@ void AsmPrinter::emitBBAddrMapSection(const MachineFunction &MF) {
     emitLabelDifferenceAsULEB128(MBB.getEndSymbol(), MBBSymbol);
     OutStreamer->emitULEB128IntValue(getBBAddrMapMetadata(MBB));
     // Emit the index of the corresponding LLVMIR basic block.
-    size_t i = 0;
-    for (auto It = F.begin(); It != F.end(); It++) {
-      const BasicBlock *BB = &*It;
-      if (BB == MBB.getBasicBlock()) {
-          break;
+    size_t BBIdx = 0;
+    bool found = false;
+    const BasicBlock *FindBB = MBB.getBasicBlock();
+    if (FindBB == nullptr) {
+      found = true;
+      BBIdx = NO_BB;
+    } else {
+      for (auto It = F.begin(); It != F.end(); It++) {
+        const BasicBlock *BB = &*It;
+        if (BB == FindBB) {
+            found = true;
```

Comment:
Surprised this wasn't caught by `clang-format`. Will need to look into that as it should flag the indentation.

Reply:
I think it's sometimes "best effort", although I agree, this one's a bit surprising!

Reply:
Are we running clang-format on the compiler itself? I didn't think so.

Reply:
Ah yes, you're right. Though I'm surprised they don't have a formatting test in one of their many checks.